core/dbt/include/global_project/macros/materializations/models/table/table.sql

```diff
@@ -31,14 +31,14 @@
     {{ get_create_table_as_sql(False, intermediate_relation, sql) }}
   {%- endcall %}
 
+  {% do create_indexes(target_relation) %}
```

Comment:
I think this will need to be
```suggestion
    {% do create_indexes(intermediate_relation) %}
```
since the table has not yet been renamed. I think that might get the failing test passing (haven't been able to check locally).